### ISSUE TITLE
principal_id = 1 in tables query can cause false circular dependency issue in privileged environment

### ISSUE THREAD

**Issue by jjm340** [jbogard/Respawn#23]
So was trying to use Respawn to run tests in an AWS RDS environment and everything works fine on my local, but every time we deployed our build and ran our tests we'd get the exception about circular dependencies.  The problem is that in RDS when run this query:
```
select s.name, t.name
from sys.tables t
INNER JOIN sys.schemas s ON t.schema_id = s.schema_id
WHERE s.principal_id = '1'
```
It would come back empty; however, the query to get relationships would bring back data.  The check for circular dependencies would then trigger.
In RDS, for whatever reason, the principal_id comes back NULL.  Does that check need to be here, I got our build working without that check, but I don't rightly know the long term consequences.  I can submit a PR, let me know.

**Comment by jbogard** [comment]
Well that's weird. I've also been told the ANSI views are bad to rely on, maybe there's a better SQL Server-specific way of doing this, are you doing SQL Server in RDS?

**Comment by jjm340** [comment]
Yeah, on my local dev machine it's a real instance, then for our CI it runs the test db in RDS.  Does that principal check need to be there?

**Comment by jbogard** [comment]
lol I have no idea it was so long ago I wrote this

**Comment by jjm340** [comment]
Well, I'd recommend taking it out and I'd submit a PR but I don't have VS 2017, right now I'm using your code modified in our app and it's working like a charm, but I'd love to stay with the latest you have going forward.

**Comment by jjm340** [comment]
@jbogard Any update on this, I'd love to keep up with your latest, but that principal condition disallows me to use this in my app in RDS.  Should I just maintain my own fork?

**Comment by jbogard** [comment]
Is there a way i can reproduce this?

**Comment by jjm340** [comment]
There is - you have to deploy a SQL Server RDS instance in AWS and attempt to use this library against it.  It will detect a circular dependency because the query above will not return any results.

**Comment by jjm340** [comment]
The reason to take it out is the reason I’ve stated. If you don’t want to,
I understand, but that means I have to maintain a fork so that it works in
my environment. All I’m looking for is a definitive statement.

**Comment by jbogard** [comment]
I'm fine taking it out, just for the life of me have no idea why that filter was there in the first place.

**Comment by jbogard** [comment]
PR?

**Comment by jjm340** [comment]
I will send a PR but I don’t have VS 2017 so I can’t run the tests or
anything - is that okay?

**Comment by jbogard** [comment]
Yeah AppVeyor will run everything from an open PR

**Comment by jjm340** [comment]
PR is up - I'm new to contributing to open source, literally this is the first time, do I need to link the issue with the PR?

**Comment by jbogard** [comment]
Yeah but that's no problem. In your comments, you can do something like "fixes #23"
